src/main/java/com/snowplowanalytics/snowplow/tracker/http/CollectorCookie.java

```diff
@@ -0,0 +1,47 @@
+package com.snowplowanalytics.snowplow.tracker.http;
+
+import okhttp3.Cookie;
+
+import java.util.*;
+
+public class CollectorCookie {
+    private final Cookie cookie;
+
+    static List<CollectorCookie> decorateAll(Collection<Cookie> cookies) {
+        List<CollectorCookie> collectorCookies = new ArrayList<>(cookies.size());
+        for (Cookie cookie : cookies) {
+            collectorCookies.add(new CollectorCookie(cookie));
+        }
+        return collectorCookies;
+    }
+
+    CollectorCookie(Cookie cookie) {
+        this.cookie = cookie;
+    }
+
+    public boolean isExpired() {
+        return cookie.expiresAt() < System.currentTimeMillis();
+    }
+
+    Cookie getCookie() {
+        return cookie;
+    }
+
+    @Override
```

Comment:
What are these `equals()` and `hashcode()` methods used for? They were in the Android `CollectorCookie` implementation

Reply:
They are standard Java methods defined in Object. `equals()` is called when you compare two objects to each other and should return true if they have the same value (not only the same reference). `hashCode()` is used by Map and Set to group objects into buckets with same/similar values. We override these methods here because we want to use the `Set<CollectorCookie>` in `CollectorCookieJar` which makes use of the functions to contain unique cookies.

[Here is a bit more discussion about the functions](https://stackoverflow.com/questions/2265503/why-do-i-need-to-override-the-equals-and-hashcode-methods-in-java).